enforcer/datapath/tokenaccessor/tokenaccessor.go

```diff
@@ -86,12 +86,13 @@ func (t *tokenAccessor) CreateAckPacketToken(context *pucontext.PUContext, auth
 // createSynPacketToken creates the authentication token
 func (t *tokenAccessor) CreateSynPacketToken(context *pucontext.PUContext, auth *connection.AuthInfo) (token []byte, err error) {
 
-	token, err = context.GetCachedToken()
-	if err == nil {
+	token, serviceContext, err := context.GetCachedTokenAndServiceContext()
```

Comment:
Maybe this should be 
token, auth.LocalServiceContext, err = context.GetCachedTokenAndServiceContext()